task-sdk/src/airflow/sdk/execution_time/secrets_masker.py

```diff
@@ -346,12 +351,29 @@ def add_mask(self, secret: str | dict | Iterable, name: str | None = None):
             for k, v in secret.items():
                 self.add_mask(v, k)
         elif isinstance(secret, str):
-            if not secret or (self._test_mode and secret in SECRETS_TO_SKIP_MASKING_FOR_TESTS):
+            if not secret:
+                return
+
+            if self._test_mode and secret.lower() in SECRETS_TO_SKIP_MASKING_FOR_TESTS:
+                return
+
+            if secret.lower() in SECRETS_TO_SKIP_MASKING:
+                return
+
+            if len(secret) < MIN_SECRET_LENGTH:
+                log.warning(
+                    "Skipping masking for a secret as it's too short (<%d chars)",
+                    MIN_SECRET_LENGTH,
+                    extra={self.ALREADY_FILTERED_FLAG: True},
+                )
```

Comment:
This warning here will issue a warning _every time_ someone pulls a connection out with a short password. I so don't think this is useful warning. Perhaps only warning _once_ per process when a short password is detected